apps/backend/api/src/modules/agent/agent.controller.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Controller,
+  Post,
+  Body,
+  Param,
+  Patch,
+  Delete,
+  Get,
+  Req,
+  ParseIntPipe,
+  BadRequestException,
+} from '@nestjs/common';
+import { UserRole } from '../../common/enums';
+import { Request } from 'express';
+import { ConfigService } from '@nestjs/config';
+import { AgentService } from './agent.service';
+import { CustomRequest } from 'src/shared/interfaces';
+
+@Controller('agents')
+export class AgentController {
+  constructor(
+    private readonly agentService: AgentService,
+    private readonly configService: ConfigService,
+  ) {}
+
+  @Post()
+  async createAgent(@Body() createAgentDto, @Req() req: CustomRequest) {
```

Comment:
```suggestion
  async create(@Body() createAgentDto, @Req() req: CustomRequest) {
```

Since this is an agent controller so there is no need to say `createAgent`